tensor/genlib/stdeng_add.go

```diff
@@ -179,24 +179,32 @@ func add{{short .}}VS(a []{{asType .}}, b {{asType .}}) {
 
 func add{{short .}}Iter(a, b []{{asType .}}, ait, bit Iterator) (err error) {	
 	var i, j int
+	var validi, validj bool
 	for {
-		if i, _, err = ait.NextValid(); err != nil {
+		if i, validi, err = ait.NextValidity(); err != nil {

```

Comment:
It might be worth thinking about this approach. NextValid() only returned at the next valid state, which left room to do future optimizations within the iterator (for example examining the mask in blocks of 16 bytes to see if any bytes are non-zero). NextValidity() as currently designed means that the iterator tests every possible state, which could make operations slower for cases with 'sparse' masks. 

For code blocks like this one, it might be quicker to stick with NextValid(), something like 
```
...
i, _, err = ait.NextValid()
j, _, err = bit.NextValid()
if i!=j{
   continue
}
...
```